src/parallel/GeneralDomainDecomposition.cpp

```diff
@@ -235,6 +237,27 @@ void GeneralDomainDecomposition::readXML(XMLfileUnits& xmlconfig) {
 	xmlconfig.getNodeValue("initialPhaseFrequency", _initFrequency);
 	global_log->info() << "GeneralDomainDecomposition frequency for initial rebalancing phase: " << _initFrequency
 					   << endl;
+
+	if(xmlconfig.changecurrentnode("loadBalancer")) {
+		std::string loadBalancerString = "None";
+		xmlconfig.getNodeValue("@type", loadBalancerString);
+		global_log->info() << "Chosen Load Balancer: " << loadBalancerString << std::endl;
+
+		std::transform(loadBalancerString.begin(), loadBalancerString.end(), loadBalancerString.begin(), ::tolower);
+
+		if (loadBalancerString.find("all") != std::string::npos) {
+			initializeALL();
+		} else {
+			global_log->error() << "GeneralDomainDecomposition: Unknown load balancer " << loadBalancerString
+			                    << ". Aborting! Please select a valid option! Valid options: ALL";
+			Simulation::exit(1);
+		}
+		_loadBalancer->readXML(xmlconfig);
+	} else{
+		global_log->error() << "loadBalancer section missing! Aborting!" << std::endl;
+	}
+	xmlconfig.changecurrentnode("..");
```

Comment:
not sure how this xml stuff works but if the `if` from line 241 fails do you still need to change the current node one up? I think this line needs to go in the true branch.

Reply:
exit missing...